### ISSUE TITLE
The MSG_WAITALL setting will block get_config() call to backend

### ISSUE THREAD

**Issue by yangzhon** [intel/cloud-hypervisor#232]
@sameo @sboeuf @rbradford 

The commit, b2f85cbdc44bb5c97a94f23409c41a33aefb2b54, which changed the recvmsg() to block mode, but this change make get_config() failed to get info from backend in user blk.

diff --git a/vhost_rs/src/vhost_user/sock_ctrl_msg.rs b/vhost_rs/src/vhost_user/sock_ctrl_msg.rs
index be9ba32..76d760f 100644
--- a/vhost_rs/src/vhost_user/sock_ctrl_msg.rs
+++ b/vhost_rs/src/vhost_user/sock_ctrl_msg.rs
@@ -176,7 +176,7 @@ fn raw_recvmsg(fd: RawFd, iovecs: &mut [iovec], in_fds: &mut [RawFd]) -> Result<

     // Safe because the msghdr was properly constructed from valid (or null) pointers of the
     // indicated length and we check the return value.
--   let total_read = unsafe { recvmsg(fd, &mut msg, 0) };
++    let total_read = unsafe { recvmsg(fd, &mut msg, libc::MSG_WAITALL) };

The config message is totally 84 bytes.
In user block, when guest kernel want to get config info from backend, the execution is stuck in recvmsg() function and do not execute to next step.

If i use the gdb to start cloud-hypersivor, this stuck can be interrupted by "ctrl+c" key and will continue to execute the left instructions.

**Comment by sboeuf** [comment]
@yangzhon 
The issue is too generic, I don't know where to start to help you with this. Could you rebase your PR #173 on top of master, now that both vhost-user fixes and vhost-user-net have been merged.
Once this is done, this will give us a sane basis to start investigating from.

Also, it seems to me the blocking behavior is caused by the amount of data sent through the socket compared to what is expected based on the header. You should check into the backend and validate it sends the appropriate amount of data, matching the header.
Here is the definition from the manual about `MSG_WAITALL`:
```
This flag requests that the operation block until the full request is satisfied.
```
And that's what we want for the configuration too. When you ask the backend to send you the configuration, you expect to wait for the completion of the request before to move forward.

**Comment by yangzhon** [comment]
@sboeuf ,

In fact, the test i did yesterday was on the latest rebase, let me renew #173 today, thanks.

**Comment by yangzhon** [comment]
@sboeuf 

one patch try to fix this issue, https://github.com/yangzhon/cloud-hypervisor/commit/9bf759e0ce6ea57015850d8472792dab2a5d7a56, please review it, thanks.

**Comment by sboeuf** [comment]
@yangzhon sorry I couldn't look at it since I've been very busy with `virtio-vsock`. I'll make sure to spend some time on it on Monday as I really want to review your code :)

**Comment by yangzhon** [comment]
This issue has been fixed in 927148dd3cfc677861d19aaad23935854b736f6a, close this issue, thanks.
